repos/keg-cli/src/utils/version/validateVersion.js

```diff
@@ -0,0 +1,36 @@
+const semver = require('semver')
+const { ask } = require('@keg-hub/ask-it')
+const { generalError } = require('../error/generalError')
+const { isValidSemver } = require('./isValidSemver')
+
+/**
+ * Validates the passed in version matches semver
+ * <br/>Checks for a valid server version
+ * <br/>Checks it is not less then the previous version
+ * @function
+ * @param {Object} publishContext - Object that defines how the repos should be published
+ * @param {string} version - New version to update to
+ * @param {string} oldVersion - Old version to update from
+ * @param {boolean} confirm - Ask to confirm the version update
+ *
+ * @returns {boolean} - If the version is valid or not
+ */
+const validateVersion = (publishContext, version, oldVersion, confirm) => {
+  // Validate the updated version
+  !isValidSemver(version) &&
+    generalError(`Invalid version ${version} for publish context ${publishContext.name}!`)
```

Comment:
Looks like `isValidSemver` already throws a general error on failure, so this isn't necessary.

Maybe rename it to `checkValidSemver`, then just call it rather than using it for boolean checking. If it throws, that code won't execute anyway.

Reply:
you're right. updated to just call `isValidSemver()`